sdk/src/main/java/com/mongodb/stitch/android/auth/oauth2/google/GoogleAuthProviderInfo.java

```diff
@@ -1,54 +1,101 @@
 package com.mongodb.stitch.android.auth.oauth2.google;
 
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.android.gms.common.api.Scope;
+import com.mongodb.stitch.android.auth.AuthProviderInfo;
+import com.mongodb.stitch.android.auth.emailpass.EmailPasswordAuthProviderInfo;
+import com.mongodb.stitch.android.auth.oauth2.facebook.FacebookAuthProviderInfo;
 
 import java.util.ArrayList;
 import java.util.List;
 
 /**
  * GoogleAuthProviderInfo contains information needed to create a {@link GoogleAuthProvider}.
  */
-public class GoogleAuthProviderInfo {
-    public static final String FQ_NAME = "oauth2/google";
+public class GoogleAuthProviderInfo extends AuthProviderInfo {
+    public static final String FQ_NAME = "oauth2-google";
+    private static final String METADATA_FIELDS = "metadata_fields";
+    private static final String CONFIG = "config";
 
-    private final String _clientId;
-    private final List<Scope> _scopes;
+    public static class MetadataField {
+        class Fields {
+            static final String NAME = "name";
+            static final String REQUIRED = "required";
+        }
 
-    @JsonCreator
-    private GoogleAuthProviderInfo(
+        private String _name;
+        private boolean _required;
 
-            @JsonProperty(Fields.CLIENT_ID)
-            final String clientId,
+        private MetadataField(@JsonProperty(Fields.NAME)
+                              @NonNull String name,
+                              @JsonProperty(Fields.REQUIRED)
+                                      boolean required) {
+            this._name = name;
+            this._required = required;
+        }
 
-            @JsonProperty(Fields.METADATA_FIELDS)
-            final List<String> scopes
-    ) {
-        _clientId = clientId;
+        @JsonProperty(Fields.NAME)
+        @NonNull
+        public String getName() {
+            return this._name;
+        }
 
-        if (scopes == null) {
-            _scopes = new ArrayList<>();
-            return;
+        @JsonProperty(Fields.REQUIRED)
+        @NonNull
+        public boolean getRequired() {
+            return this._required;
         }
+    }
 
-        final List<Scope> gScopes = new ArrayList<>(scopes.size());
-        for (final String scope : scopes) {
-            gScopes.add(new Scope(scope));
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class Config {
+        static class Fields {
+            static final String CLIENT_ID = "clientId";
```

Comment:
here too?

Reply:
Unfortunately we didn't update these to snakecase on the serverside